### PR TITLE
Issue #126: Update schedule command to check for candidate availability

### LINKED ISSUE #126: Update `schedule` command to check for Candidate's availabilities
Currently, the `schedule` command implements basic functionality which schedules the candidate's interview simple based on the date and time entered by the user.

This function of `schedule` will be updated to implement checks on whether the scheduled date and time falls on a day that the candidate has indicated availability for.

### LINKED ISSUE #141: Update `schedule` command to check if interview is on a weekday and between 8am - 6pm
This function of schedule will be updated to implement checks on whether the scheduled time falls on a week day and is between 8am - 6pm (i.e. last interview should be at 5.30pm)

### PR DESCRIPTION
**Pull request by charmainehly**
Currently, the schedule command does not check whether the user's input for interview datetime matches the candidate's availability.

This allows interviews to be set for candidates on days that the candidates are not available. Instead, this should be invalid
and should throw an error.

Let's implement checks to ensure the interview datetime matches the candidate's availabilities. A new method through the
ModelManager and InterviewSchedule class will return true if the interview datetime matches the candidate's availability, and false otherwise. The schedule command will throw an error message if there is no match.

Closes #126 

Update: Functionality for checking that the interview is being scheduled during office hours has also been implemented within this branch. Closes #141

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ScheduleCommand.java [hunk 1]
```diff
@@ -77,6 +80,10 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_CONFLICTING_INTERVIEW);
         }
 
+        if (!model.hasMatchingAvailability(toAdd)) {
+            throw new CommandException(MESSAGE_CANDIDATE_NOT_AVAILABLE);
+        }
```
Comment (lzan98) [inline src/main/java/seedu/address/logic/commands/ScheduleCommand.java:85]: Nice work Charmaine! Just a food for thought to consider, is it necessary to go through the entire Model/InterviewSchedule/UniqueInterviewList if we are not checking against other interviews!

**Review by lzan98 — APPROVED** [review]
Nice work! Left (quite a big) suggestion for you to consider, can be done in future (or current) PR if you agree with it! Otherwise just ignore it hehe.

**Comment by charmainehly** [comment]
Update: Closes #126 and Closes #141

File: src/main/java/seedu/address/logic/commands/ScheduleCommand.java (hunk 1 above)
Comment (charmainehly) [reply src/main/java/seedu/address/logic/commands/ScheduleCommand.java:90]: Thanks for reviewing the implementation! :) I relooked at it, and I agree, I think the layers of abstraction was unnecessary, I've instead made the implementation for the methods within the `Interview` class. Let me finalise the additional commits before re-requesting for approval!

**Review by lzan98 — APPROVED** [review]
LGTM! Nice test cases too.

**Pull request merged**